### PR TITLE
ES-2378: REST client: remove MGMApi workaround

### PR DESCRIPTION
**Pull request by anton-subbotin**
For the endpoints which return `String` we need to make generated REST client return `.body()` as a String instead of deserialising with ObjectMapper.

See details in Jira: https://r3-cev.atlassian.net/browse/ES-2378?focusedCommentId=306841

### CODE REVIEW

File: libs/rest/generated-rest-client/build.gradle [hunk 1]
```diff
@@ -133,45 +133,18 @@ tasks.register('applyWorkaroundForCertificateApi') {
...
-                .replaceAll(targetLine1, replacementLine1)
-                .replaceAll(targetLine2, replacementLine2)
-    }
-}
-
-tasks.register('applyWorkaroundForApiClient') {
-    dependsOn 'openApiGenerate'
-    doLast {
-        def targetLine1 = 'Serializer.jacksonObjectMapper.readValue\\(bodyContent, object: TypeReference<T>\\(\\) \\{}\\)'
+        def targetLine = 'Serializer.jacksonObjectMapper.readValue\\(bodyContent, object: TypeReference<T>\\(\\) \\{}\\)'
```
Comment (tlawson3) [inline libs/rest/generated-rest-client/build.gradle]: Will you add the task input/output/uptoDateWhen, like in this draft PR: https://github.com/corda/corda-runtime-os/pull/6174/files

It makes the task more complete

File: libs/rest/generated-rest-client/src/test/kotlin/net/corda/restclient/generated/TestKnownIssues.kt [hunk 2]
```diff
@@ -151,5 +159,4 @@ class TestKnownIssues {
         assertThat(roles).isNotEmpty
         assertThat(roles.first().updateTimestamp).isNotNull
     }
-
-}
\ No newline at end of file
+}
```
Comment (tlawson3) [inline libs/rest/generated-rest-client/src/test/kotlin/net/corda/restclient/generated/TestKnownIssues.kt:243]: Can you add a test for that SQL endpoint you found?

**Review by tlawson3 — CHANGES_REQUESTED** [review]

**Review by tlawson3 — DISMISSED** [review]

File: libs/rest/generated-rest-client/src/test/kotlin/net/corda/restclient/generated/TestKnownIssues.kt (hunk 2 above)
Comment (anton-subbotin) [reply libs/rest/generated-rest-client/src/test/kotlin/net/corda/restclient/generated/TestKnownIssues.kt:243]: There are 10 overall endpoints with String return type. For now we are covering two of them - get MGM GroupPolicy and post CSR request; do you still think we need to add more?

Comment (anton-subbotin) [reply libs/rest/generated-rest-client/src/test/kotlin/net/corda/restclient/generated/TestKnownIssues.kt:243]: If we want to be 100% safe we need to add tests for all of them...

Comment (tlawson3) [reply libs/rest/generated-rest-client/src/test/kotlin/net/corda/restclient/generated/TestKnownIssues.kt:243]: These tests are mock tests, which aren't actually triggering Corda. All we're covering is the generated client's ability to handle the expected response. My question is, are these 10 endpoints providing the similar kind of response to one another so we could assume we're ok? The SQL value could be different, depending on how it's escaped or not.

Comment (anton-subbotin) [reply libs/rest/generated-rest-client/src/test/kotlin/net/corda/restclient/generated/TestKnownIssues.kt:243]: All of the responses are similar in a sense that it is a String with whatever a String can be, and are not supposed to be deserialised - even in the case of MGM Group Policy which _is_ a valid JSON String. 

I'll go and add the tests just to be on a safe side.

But then, ideally, we'd better have mock tests for all existing possible endpoints..

File: tools/plugins/network/src/main/kotlin/net/corda/cli/plugins/network/BaseOnboard.kt [hunk 3]
```diff
@@ -227,7 +227,7 @@ abstract class BaseOnboard : Runnable, RestCommand() {
             throw OnboardException("Could not submit registration request: ${response.memberInfoSubmitted}")
         }
 
-        println("Registration ID for '$name' is '$registrationId'")
+        println("Registration ID for '$name' is '${registrationId.value}'")
```
Comment (anton-subbotin) [inline tools/plugins/network/src/main/kotlin/net/corda/cli/plugins/network/BaseOnboard.kt:230]: Spotted this one when running pluginSmokeTest

File: libs/rest/generated-rest-client/src/test/kotlin/net/corda/restclient/generated/TestKnownIssues.kt (hunk 2 above)
Comment (anton-subbotin) [reply libs/rest/generated-rest-client/src/test/kotlin/net/corda/restclient/generated/TestKnownIssues.kt:243]: added few more tests for different kinds of String responses (PEM, greeting, SQL)

**Review by tlawson3 — APPROVED** [review]

**Review by ronanbrowne — APPROVED** [review]
LGTM (build changes)

**Pull request merged**